lib/msal-node/test/cache/TokenCache.spec.ts

```diff
@@ -246,4 +246,68 @@ describe("TokenCache tests", () => {
             }
         );
     });
+
+    it("overwriteCache should overwrite the in-memory cache with persistent cache if exists", async () => {
+        const cachePath = "./test/cache/cache-test-files/default-cache.json";
+        const beforeCacheAccess = async (context: TokenCacheContext) => {
+            context.tokenCache.deserialize(
+                await promises.readFile(cachePath, "utf-8")
+            );
+        };
+        const afterCacheAccess = async (context: TokenCacheContext) => {
+            await promises.writeFile(cachePath, context.tokenCache.serialize());
+        };
+
+        const cachePlugin: ICachePlugin = {
+            beforeCacheAccess,
+            afterCacheAccess,
+        };
+
+        const tokenCache = new TokenCache(storage, logger, cachePlugin);
+
+        const mockTokenCacheContextInstance = {
+            hasChanged: false,
+            cache: tokenCache,
+            cacheHasChanged: false,
+            tokenCache,
+        };
+
+        jest.spyOn(msalCommon, "TokenCacheContext").mockImplementation(
+            () => mockTokenCacheContextInstance as unknown as TokenCacheContext
+        );
+
+        const clearSpy = jest.spyOn(NodeStorage.prototype, "clear");
+        const setSpy = jest.spyOn(NodeStorage.prototype, "setCache");
+        const snapshotSpy = jest.spyOn(
+            TokenCache.prototype,
+            "getCacheSnapshot"
+        );
+
+        await tokenCache.overwriteCache();
+        // overwriting means both clearing and setting the in-memory cache to the persistent cache, we check for both
+        expect(clearSpy).toHaveBeenCalled();
+        expect(setSpy).toHaveBeenCalledWith(snapshotSpy.mock.results[0].value);
```

Comment:
Can you and/or check the value of `tokenCache.getCacheSnapshot` to check the value of the cache to make sure it was overridden?